src/pydata_sphinx_theme/assets/styles/sections/_sidebar-primary.scss

```diff
@@ -140,61 +141,107 @@
   .list-caption {
     list-style: none;
     padding-left: 0px;
-  }
-  li {
-    position: relative;
-    // If it has children, add a bit more padding to wrap the content to avoid
-    // overlapping with the <label>
-    &.has-children {
-      > .reference {
-        padding-right: 30px;
+
+    // Level 0 TOC heading is put inside the <summary> tag
+    // so let the <summary> tag take up more space
+    li.toctree-l0.has-children {
+      > details {
+        > summary {
+          position: relative;
+          height: auto;
+          width: auto;
+          display: flex;
+          justify-content: space-between;
+          align-items: baseline;
+
+          .toctree-toggle {
+            // Prevent toggle icon from getting squished by summary being a
+            // flexbox
```

Comment:
just want to say I really appreciate comments like this in the CSS. Makes reviewing so much easier when the intent is clear.